### PR TITLE
Twin naming and enum option update

### PR DESCRIPTION
**Pull request by momuno**
- Changes naming to Twin only.
- Changes option from using a string literal to an enum.  Updates logic to build username.

### CODE REVIEW

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 1]
```diff
@@ -26,18 +26,6 @@
 
 #include <azure/core/_az_cfg_prefix.h>
 
-/**
```
Comment (ericwol-msft) [inline sdk/inc/azure/iot/az_iot_hub_client.h:29]: as a comment from Dane other PR. why was this removed?

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 2]
```diff
@@ -47,6 +35,16 @@ enum
   AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_QOS = 0
 };
 
+/**
+ * @brief Azure IoT Hub Client twin_content_type option values.
+ *
+ */
+typedef enum
+{
+  AZ_IOT_HUB_CLIENT_OPTION_TWIN_CONTENT_TYPE_DEFAULT_JSON = 0,
```
Comment (ericwol-msft) [inline sdk/inc/azure/iot/az_iot_hub_client.h:44]: I like it!!!

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 3]
```diff
@@ -47,6 +35,16 @@ enum
   AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_QOS = 0
 };
 
+/**
+ * @brief Azure IoT Hub Client twin_content_type option values.
```
Comment (ericwol-msft) [inline sdk/inc/azure/iot/az_iot_hub_client.h:39]:
```suggestion
 * @brief Azure IoT Hub Client `twin_content_type` option values.
```

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 4]
```diff
@@ -63,18 +61,19 @@ typedef struct
...
-  az_span method_twin_content_type;
-
   /**
    * The `model_id` is used to identify the capabilities of a device based on the Digital Twin
    * Document.
    */
   az_span model_id;
+
+  /**
+   * The `twin_content_type` determines the format to send in the username, which will inform IoT
```
Comment (ericwol-msft) [inline sdk/inc/azure/iot/az_iot_hub_client.h:71]:
```suggestion
   * The `twin_content_type` determines the format to send in the username filed properties, which will inform the IoT
```

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 5]
```diff
@@ -63,18 +61,19 @@ typedef struct
...
+   * Hub what format the device expects the twin document to use. If this option is not set, the
```
Comment (ericwol-msft) [inline sdk/inc/azure/iot/az_iot_hub_client.h:72]:
```suggestion
   * Hub what format the device expects the twin document encoding. If this option is not set, the
```

File: sdk/src/azure/iot/az_iot_hub_client.c [hunk 6]
```diff
@@ -23,15 +23,16 @@ static const az_span hub_service_api_version = AZ_SPAN_LITERAL_FROM_STR("/?api-v
 static const az_span hub_client_sdk_version
     = AZ_SPAN_LITERAL_FROM_STR("DeviceClientType=c%2F" AZ_SDK_VERSION_STRING);
 static const az_span hub_digital_twin_model_id = AZ_SPAN_LITERAL_FROM_STR("model-id");
-static const az_span hub_method_twin_content_type
-    = AZ_SPAN_LITERAL_FROM_STR("default-content-type");
+static const az_span hub_twin_content_type = AZ_SPAN_LITERAL_FROM_STR("default-content-type");
+static const az_span hub_twin_content_type_cbor = AZ_SPAN_LITERAL_FROM_STR("application%2Fcbor");
```
Comment (ericwol-msft) [inline sdk/src/azure/iot/az_iot_hub_client.c:27]: from your sample, looks like the app also needs this....move back to the header?

**Review by ericwol-msft — APPROVED** [review]

File: sdk/src/azure/iot/az_iot_hub_client.c (hunk 6 above)
Comment (momuno) [reply sdk/src/azure/iot/az_iot_hub_client.c:27]: Curious where do you see it needing it?  If for telemetry or c2d, that is application defined -- not to be conflated with twin username.

File: sdk/inc/azure/iot/az_iot_hub_client.h (hunk 5 above)
Comment (momuno) [reply sdk/inc/azure/iot/az_iot_hub_client.h:72]: same per comment in other repo on encoding.

File: sdk/inc/azure/iot/az_iot_hub_client.h (hunk 3 above)
Comment (momuno) [reply sdk/inc/azure/iot/az_iot_hub_client.h:39]: good eye
